File: game_client/Assets/Scripts/ClientProtocol/GS2GC/p023_ArenaOp/GS2GC_023_021_RetTowerFight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;


namespace GS2GC.p023_ArenaOp
{

public class GS2GC_023_021_RetTowerFight : ALBasicProtocolPack._IALProtocolStructure {
/// <summary>
/// 是否打败
/// </summary>
private bool isDefeat;
/// <summary>
/// 奖励列表
/// </summary>
private List<NPCommon.NPCommon_ItemInfo> rewardList;
/// <summary>
/// 玩家ID 如果为0则为NPC
/// </summary>
private long cid;


public GS2GC_023_021_RetTowerFight() {
	isDefeat = false;
	rewardList = new List<NPCommon.NPCommon_ItemInfo>();
	cid = (long)0;
}

public GS2GC_023_021_RetTowerFight(
	bool _isDefeat
	, List<NPCommon.NPCommon_ItemInfo> _rewardList
	, long _cid
) {	isDefeat = _isDefeat;
	rewardList = _rewardList;
	cid = _cid;
}

public byte getMainOrder() { return (byte)23; }

public byte getSubOrder() { return (byte)21; }

/// <summary>
/// 是否打败
/// </summary>
public bool getIsDefeat() { return isDefeat; }
/// <summary>
/// 是否打败
/// </summary>
public void setIsDefeat(bool _isDefeat) { isDefeat = _isDefeat; }
/// <summary>
/// 奖励列表
/// </summary>
public List<NPCommon.NPCommon_ItemInfo> getRewardList() { return rewardList; }
/// <summary>
/// 奖励列表
/// </summary>
public void addRewardList(NPCommon.NPCommon_ItemInfo _rewardList) { rewardList.Add(_rewardList); }
/// <summary>
/// 玩家ID 如果为0则为NPC
/// </summary>
public long getCid() { return cid; }
/// <summary>
/// 玩家ID 如果为0则为NPC
/// </summary>
public void setCid(long _cid) { cid = _cid; }


public int GetBufSize() {
	int _size = 9;
	_size += 2;
for(int _i = 0; _i < rewardList.Count; _i++) {
	_size += 4 + rewardList[_i].GetBufSize();
	}


	return _size;
}

public int GetFullPackBufSize() {
	int _size = 11;
	_size += 2;
for(int _i = 0; _i < rewardList.Count; _i++) {
	_size += 4 + rewardList[_i].GetBufSize();
	}


	return _size;
}



public void ReadUnzipBuf(ALProtocolBuf _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	isDefeat = (_buf.get() != 0);
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	short _rewardListCount = _buf.getShort();
	for(int _i = 0; _i < _rewardListCount; _i++) { 
		NPCommon.NPCommon_ItemInfo _rewardList = new NPCommon.NPCommon_ItemInfo();
		int __rewardListCustLen = _buf.getInt();
	int __rewardListCurPos = _buf.getCurPos();
	_rewardList.ReadUnzipBuf(_buf, __rewardListCurPos + __rewardListCustLen);
	_buf.setPosition(__rewardListCurPos + __rewardListCustLen);

		rewardList.Add(_rewardList);
	}
	 if(_finalPos > 0 && _buf.getCurPos() >= _finalPos) return ;
	cid = _buf.getLong();
}

public void PutUnzipBuf(ALProtocolBuf _buf) {
	_buf.put(isDefeat?(byte)1:(byte)0);
	_buf.putShort((short)rewardList.Count);
	for(int _i = 0; _i < rewardList.Count; _i++) { 
		_buf.putInt(rewardList[_i].GetBufSize());
	rewardList[_i].PutUnzipBuf(_buf);
	}
	_buf.putLong(cid);
}

public byte[] makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	_buf.put((byte)23);
	_buf.put((byte)21);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void makeFullPackage(ALProtocolBuf _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)23);
	_recBuf.put((byte)21);
	PutUnzipBuf(_recBuf);
}
public byte[] makePackage() {
	int _bufSize = GetBufSize();
	ALProtocolBuf _buf = ALProtocolBuf.allocate(_bufSize);
	PutUnzipBuf(_buf);
	return _buf.getBuf();
}
public void readPackage(byte[] _buf) {
	ALProtocolBuf _bufObj = new ALProtocolBuf(_buf);
	ReadUnzipBuf(_bufObj, -1);
}
public void readPackage(ALProtocolBuf _buf) {
	ReadUnzipBuf(_buf, -1);
}
public override string ToString() {
	System.Text.StringBuilder builder = new System.Text.StringBuilder();

	builder.Append("{");
	builder.Append("isDefeat").Append(":").Append(isDefeat.ToString()).Append(", ");
	builder.Append("rewardList").Append(":").Append(rewardList.ToString()).Append(", ");
	builder.Append("cid").Append(":").Append(cid.ToString()).Append(", ");
	builder.Append("}");
	return builder.ToString();
}

}

}

File: game_server/ServerProtocol/src/GS2GC/p023_ArenaOp/GS2GC_023_021_RetTowerFight.java
package GS2GC.p023_ArenaOp;

import java.nio.ByteBuffer;
public class GS2GC_023_021_RetTowerFight implements ALBasicProtocolPack._IALProtocolStructure {
/** 是否打败 */
private boolean isDefeat;
/** 奖励列表 */
private java.util.ArrayList<NPCommon.NPCommon_ItemInfo> rewardList;
/** 玩家ID 如果为0则为NPC */
private long cid;


public GS2GC_023_021_RetTowerFight() {
	isDefeat = false;
	rewardList = new java.util.ArrayList<NPCommon.NPCommon_ItemInfo>();
	cid = (long)0;
}

public GS2GC_023_021_RetTowerFight(
	 boolean _isDefeat
	, java.util.ArrayList<NPCommon.NPCommon_ItemInfo> _rewardList
	, long _cid
) {	isDefeat = _isDefeat;
	rewardList = _rewardList;
	cid = _cid;
}

public final byte getMainOrder() { return (byte)23; }

public final byte getSubOrder() { return (byte)21; }

/** 是否打败 */
public boolean getIsDefeat() { return isDefeat; }
/** 是否打败 */
public void setIsDefeat(boolean _isDefeat) { isDefeat = _isDefeat; }
/** 奖励列表 */
public java.util.ArrayList<NPCommon.NPCommon_ItemInfo> getRewardList() { return rewardList; }
/** 奖励列表 */
public void addRewardList(NPCommon.NPCommon_ItemInfo _rewardList) { rewardList.add(_rewardList); }
/** 玩家ID 如果为0则为NPC */
public long getCid() { return cid; }
/** 玩家ID 如果为0则为NPC */
public void setCid(long _cid) { cid = _cid; }


public final int GetBufSize() {
	int _size = 9;
	_size += 2;
	for(int _i = 0; _i < rewardList.size(); _i++) {
	_size += 4 + rewardList.get(_i).GetBufSize();
	}


	return _size;
}

public final int GetFullPackBufSize() {
	int _size = 11;
	_size += 2;
	for(int _i = 0; _i < rewardList.size(); _i++) {
	_size += 4 + rewardList.get(_i).GetBufSize();
	}


	return _size;
}



public final void ReadUnzipBuf(ByteBuffer _buf, int _finalPos) {
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) isDefeat = (_buf.get() != 0);
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() <= 0) return;
	short _rewardListCount = _buf.getShort();
	for(int _i = 0; _i < _rewardListCount; _i++) { 
		NPCommon.NPCommon_ItemInfo _rewardList = new NPCommon.NPCommon_ItemInfo();
		if(_buf.remaining() <= 0) return;
	int __rewardListCustLen = _buf.getInt();
	int __rewardListCurPos = _buf.position();
	_rewardList.ReadUnzipBuf(_buf, __rewardListCurPos + __rewardListCustLen);
	_buf.position(__rewardListCurPos + __rewardListCustLen);

		rewardList.add(_rewardList);
	}
	 if(_finalPos > 0 && _buf.position() >= _finalPos) return ;
	if(_buf.remaining() > 0) cid = _buf.getLong();
}

public final void PutUnzipBuf(ByteBuffer _buf) {
	_buf.put(isDefeat?(byte)1:(byte)0);
	_buf.putShort((short)rewardList.size());
	for(int _i = 0; _i < rewardList.size(); _i++) { 
		_buf.putInt(rewardList.get(_i).GetBufSize());
	rewardList.get(_i).PutUnzipBuf(_buf);
	}
	_buf.putLong(cid);
}

public final ByteBuffer makeFullPackage() {
	int _bufSize = GetBufSize() + 2;
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	_buf.put((byte)23);
	_buf.put((byte)21);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void makeFullPackage(ByteBuffer _recBuf) {
	if(null == _recBuf)
		return ;
	_recBuf.put((byte)23);
	_recBuf.put((byte)21);
	PutUnzipBuf(_recBuf);
}
public final ByteBuffer makePackage() {
	int _bufSize = GetBufSize();
	ByteBuffer _buf = ByteBuffer.allocate(_bufSize);
	PutUnzipBuf(_buf);
	_buf.flip();
	return _buf;
}
public final void readPackage(ByteBuffer _buf) {
	ReadUnzipBuf(_buf, -1);
}
}

